src/shogun/mathematics/linalg/ratapprox/logdet/LogDetEstimator.cpp

```diff
@@ -145,72 +122,29 @@ SGVector<float64_t> CLogDetEstimator::sample(index_t num_estimates)
 		m_operator_log->get_operator()->get_dimension(),
 		m_trace_sampler->get_dimension());
 
-	// for storing the aggregators that submit_jobs return
-	CDynamicObjectArray* aggregators=new CDynamicObjectArray();
 	index_t num_trace_samples=m_trace_sampler->get_num_samples();
-
-	for (index_t i=0; i<num_estimates; ++i)
-	{
-		for (index_t j=0; j<num_trace_samples; ++j)
-		{
-			SG_INFO("Computing log-determinant trace sample %d/%d\n", j,
-					num_trace_samples);
-
-			SG_DEBUG("Creating job for estimate %d, trace sample %d/%d\n", i, j,
-					num_trace_samples);
-			// get the trace sampler vector
-			SGVector<float64_t> s=m_trace_sampler->sample(j);
-			// create jobs with the sample vector and store the aggregator
-			CJobResultAggregator* agg=m_operator_log->submit_jobs(s);
-			aggregators->append_element(agg);
-			SG_UNREF(agg);
-		}
-	}
-
-	REQUIRE(m_computation_engine, "Computation engine is NULL\n");
-
-	// wait for all the jobs to be completed
-	SG_INFO("Waiting for jobs to finish\n");
-	m_computation_engine->wait_for_all();
-	SG_INFO("All jobs finished, aggregating results\n");
-
-	// the samples vector which stores the estimates with averaging
 	SGVector<float64_t> samples(num_estimates);
 	samples.zero();
-
-	// use the aggregators to find the final result
-	// use the same order as job submission to combine results
-	int32_t num_aggregates=aggregators->get_num_elements();
-	index_t idx_row=0;
-	index_t idx_col=0;
-	for (int32_t i=0; i<num_aggregates; ++i)
-	{
-		// this cast is safe due to above way of building the array
-		CJobResultAggregator* agg=dynamic_cast<CJobResultAggregator*>
-			(aggregators->get_element(i));
-		ASSERT(agg);
-
-		// call finalize on all the aggregators, cast is safe again
-		agg->finalize();
-		CScalarResult<float64_t>* r=dynamic_cast<CScalarResult<float64_t>*>
-			(agg->get_final_result());
-		ASSERT(r);
-
-		// iterate through indices, group results in the same way as jobs
-		samples[idx_col]+=r->get_result();
-		idx_row++;
-		if (idx_row>=num_trace_samples)
+	// for omp
```

Comment:
don't put those comments in, they don't add information :)

(as I said, we can postpone the openmp into a new PR to make things easier to handle if you wanted to)